system_tests/multigas_stylus_program_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+package arbtest
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/offchainlabs/nitro/arbos/multigascollector"
+	"github.com/offchainlabs/nitro/util/testhelpers"
+)
+
+func TestMultigasStylus_GetBytes32(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	outDir := t.TempDir()
+
+	// Build a node with the multigas collector enabled
+	builder := NewNodeBuilder(ctx).
+		DefaultConfig(t, false).
+		WithMultigasCollector(multigascollector.CollectorConfig{
+			OutputDir:      outDir,
+			BatchSize:      5,
+			ClearOutputDir: true,
+		})
+	cleanup := builder.Build(t) // no defer
+
+	l2info := builder.L2Info
+	l2client := builder.L2.Client
+
+	// Deploy programs
+	owner := l2info.GetDefaultTransactOpts("Owner", ctx)
+	storage := deployWasm(t, ctx, owner, l2client, rustFile("storage"))
+
+	// Send tx to call getBytes32
+	key := testhelpers.RandomHash()
+	readArgs := argsForStorageRead(key)
+	tx := l2info.PrepareTxTo("Owner", &storage, l2info.TransferGas, nil, readArgs)
+	require.NoError(t, l2client.SendTransaction(ctx, tx))
+	_, err := EnsureTxSucceeded(ctx, l2client, tx)
+	require.NoError(t, err)
+
+	// Stop node to flush collector
+	cleanup()
+
+	var blocks = readCollectorBatches(t, outDir, -1)
+	require.NotEmpty(t, blocks, "no multigas data found")
+
+	// Find tx in last block and assert StorageAccess was charged
+	txHash := tx.Hash().Bytes()
+	var found bool
+	for _, ptx := range blocks[len(blocks)-1].Transactions {
+		if !bytes.Equal(ptx.TxHash, txHash) {
+			continue
+		}
+		require.NotNil(t, ptx.MultiGas, "missing multigas for tx %s", tx.Hash())
+
+		require.Equal(t, uint64(params.ColdSloadCostEIP2929), ptx.MultiGas.StorageAccess)
+		require.Greater(t, ptx.MultiGas.WasmComputation, uint64(0))
```

Comment:
Can you check the exact value here? This should be deterministic.

Reply:
I tried it and it seems to me that this is not quite the right way to approach it. The price is made up of many components, such as the ArbWasm precompile for gas params, program init/cached gas costs, and `evmMemoryCost` (a private function from `arbos/programs`). Even if I mirror these calculations in the test, it will be fragile, because any change in the calculation logic or inputs would require redoing the test, even though the code we’re testing hasn’t changed. My opinion is that we should rely on the total gas to test only the changed logic in CallProgram. But since there is no total gas in the collector yet, I left a TODO and I’ll fix it within NIT-3767. Is this fine?

Reply:
My idea was to run the code once, get the result, and hardcode the expected value. Doing that would ensure we don't have regressions in the future. But we can also do that later, after 3767.

Reply:
Just noticed, on CI the value is different:
```
multigas_stylus_program_test.go:73: 
        	Error Trace:	/home/runner/work/nitro/nitro/system_tests/multigas_stylus_program_test.go:73
        	Error:      	Not equal: 
        	            	expected: 0x28b7 (10423)
        	            	actual  : 0x2f88 (12168)
        	Test:       	TestMultigasStylus_GetBytes32
```
@gligneul are you ok leaving it like this?
```
			// TODO: Once all operations are instrumented, WasmComputation
			// should be derived as the residual from SingleGas instead of asserted directly.
			require.Greater(t, ptx.MultiGas.WasmComputation, uint64(0))
```